### ISSUE TITLE
"comparing identical expressions" discourages using un-broken isNaN() substitute

### ISSUE THREAD

**Issue by yahelc** [douglascrockford/JSLint#66]
Working from the notions here, https://gist.github.com/1086528 , isNaN() is problematic because it coerces its value to a number. The best way to explicitly identify something as having the value of `NaN` is to do a reflexive comparison, since `NaN` is the only value for which `x==x` returns false. 

ie,

```
function betterIsNaN(x){
    return !(x===x);
}
```

Except, JSLint throws a warning when I use that: 

```
62: WARNING: useless comparison; comparing identical expressions
```

It's not quite accurate, since the potential for NaN values in the comparison means that identical values could potentially result in a reflexive expression evaluating as false. 

Can you look at removing this warning, since this seems a valid application of comparing identical expressions?
